examples/platform/silabs/efr32/wf200/efr_spi.c

```diff
@@ -49,14 +48,24 @@
 #include "sl_power_manager.h"
 #endif
 
-#if SL_WIFI
+#if defined(EFR32MG12)
+#include "sl_spidrv_exp_config.h"
+extern SPIDRV_Handle_t sl_spidrv_exp_handle;
+#define SL_SPIDRV_HANDLE sl_spidrv_exp_handle
+#elif defined(EFR32MG24)
+#include "sl_spidrv_eusart_exp_config.h"
 #include "spi_multiplex.h"
-StaticSemaphore_t spi_sem_peripharal;
-SemaphoreHandle_t spi_sem_sync_hdl;
+#else
+#error "Unknown platform"
 #endif
 
 #define USART SL_WFX_HOST_PINOUT_SPI_PERIPHERAL
 
+#if defined(EFR32MG24)
+StaticSemaphore_t spi_sem_peripheral;
```

Comment:
having variable encased inside of a `#ifdef` seems like a bad way to go. Wasn't there another solution that would have been streamlined/common for both chips ? MG12 & MG24?